cfme/tests/infrastructure/test_vm_reconfigure.py

```diff
@@ -196,34 +196,68 @@ def test_vm_reconfig_add_remove_hw_hot(provider, small_vm, ensure_vm_running, ch
     reconfigure_vm(small_vm, orig_config)
 
 
-@pytest.mark.manual
+@pytest.mark.rhel_testing
 @pytest.mark.tier(1)
-@pytest.mark.parametrize('change_type', ['sockets', 'memory'])
-def test_vm_reconfig_add_remove_hw_hot_vmware(change_type):
+@pytest.mark.rhv1
+@pytest.mark.parametrize('disk_type', ['thin', 'thick'])
+@pytest.mark.parametrize(
+    'disk_mode', ['persistent', 'independent_persistent', 'independent_nonpersistent'])
+@pytest.mark.uncollectif(
+    # Disk modes cannot be specified when adding disk to VM in RHV provider
+    lambda disk_mode, provider: disk_mode != 'persistent' and provider.one_of(RHEVMProvider))
+def test_vm_reconfig_add_remove_disk_hot(small_vm, disk_type, disk_mode):
+
     """
     Polarion:
         assignee: nansari
         initialEstimate: 1/6h
         testtype: functional
-        startsin: 5.5
+        startsin: 5.9
         casecomponent: Infra
         tags: reconfigure
         testSteps:
-            1. Change number of CPU sockets and amount of memory while VM is runnng.
+            1. Add and remove the disk while VM is running
             2. Go to Compute -> infrastructure -> Virtual Machines -> Select Vm
             3. Go to VM reconfiguration
-            4.Change number of CPU sockets and amount of memory, save and submit
+            4. Click on Add Disk -> select disk_type and disk_mode , save and submit
             5. Check the count in VM details page
+            6. Remove the disk and Check the count in VM details page
     """
-    pass
+    orig_config = small_vm.configuration.copy()
+    new_config = orig_config.copy()
+    new_config.add_disk(
+        size=500, size_unit='MB', type=disk_type, mode=disk_mode
+    )
+
+    add_disk_request = small_vm.reconfigure(new_config)
+    # Add disk request verification
+    wait_for(add_disk_request.is_succeeded, timeout=360, delay=15,
+             message="confirm that disk was added")
+    # Add disk UI verification
+    wait_for(
+        lambda: small_vm.configuration.num_disks == new_config.num_disks, timeout=360, delay=15,
+        fail_func=small_vm.refresh_relationships,
+        message="confirm that disk was added")
+    msg = "Disk wasn't added to VM config"
+    assert small_vm.configuration.num_disks == new_config.num_disks, msg
+
+    remove_disk_request = small_vm.reconfigure(orig_config)
+    # Remove disk request verification
+    wait_for(remove_disk_request.is_succeeded, timeout=360, delay=15,
+             message="confirm that previously-added disk was removed")
+    # Remove disk UI verification
+    wait_for(
+        lambda: small_vm.configuration.num_disks == orig_config.num_disks, timeout=360, delay=15,
+        fail_func=small_vm.refresh_relationships,
+        message="confirm that previously-added disk was removed")
+    msg = "Disk wasn't removed from VM config"
+    assert small_vm.configuration.num_disks == orig_config.num_disks, msg
```

Comment:
Just like @john-dupuy commented above, if you're not going to use `msg` anywhere else, you might directly want to use it in the `assert` statement, like this:
` assert small_vm.configuration.num_disks == orig_config.num_disks, "Disk wasn't removed from VM config" `

I think this is the practice that we have been following in our codebase.

Reply:
@valaparthvi yes it would be good.